satpy/modifiers/atmosphere.py

```diff
@@ -45,18 +45,17 @@ def __call__(self, projectables, optional_datasets=None, **info):
         else:
             vis, red, sata, satz, suna, sunz = self.match_data_arrays(
                 projectables + optional_datasets)
+            # First make sure the two azimuth angles are in the range 0-360:
+            sata = sata % 360.
+            suna = suna % 360.
```

Comment:
Doesn't this being here require #2166?

Reply:
I don't think so, no. My testing for this (mainly with GOES rayleigh + aerosol correction) is fine without #2166.

Reply:
My worry is that this is now under only this `if` statement and is needed if things aren't on a 0-360 range. Your #2166 specifically changes the range of the functions in `angles.py` so I would expect this to fail for GOES-R (or any angles.py generated angles). If it doesn't, then that suggests that something is doing the modulo somewhere else and it might not be needed here at all.

Reply:
Aaah, so you'd rather it was outside the `if`, to catch 'bad' angles in all circumstances? I can move it outside.

Reply:
Well you specifically moved it under this `else` block which was my preferred placement *after* #2166 is merged, but until that happens I think this placement in this PR (and without #2166) will produce invalid output for cases where angles are generated (not provided by the reader) and have values between -180 and 0.

Reply:
No, the output is valid - I tested it (without #2166 for GOES and the results are as expected). Can double check and post them here if you'd like.

Reply:
If you are sure that the test case you used included the angles with -180 to 0 values then...I'm confused. I would have expected it to fail.

Reply:
Or maybe because of all the trigonometry and `2 * np.pi` usage in pyorbital it "just works"? If not already that should maybe be explicitly tested in pyorbital and documented.

Reply:
Right, done that!